mbuild/compound.py

```diff
@@ -312,8 +323,205 @@ def particles_by_name(self, name):
+        a rigid body
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.label_rigid_bodies(rigid_particles='C')
+
+        Create a box of rigid benzenes, where each benzene has a unique rigid
+        body ID.
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.name = 'Benzene'
+        >>> filled = mb.fill_box(benzene,
+        ...                      n_compounds=10,
+        ...                      box=[0, 0, 0, 4, 4, 4])
+        >>> filled.label_rigid_bodies(distinct_bodies='Benzene')
+
+        Create a box of semi-rigid benzenes, where each benzene has a unique
+        rigid body ID and only the carbon portion is treated as rigid.
+
+        >>> import mbuild as mb
+        >>> from mbuild.utils.io import get_fn
+        >>> benzene = mb.load(get_fn('benzene.mol2'))
+        >>> benzene.name = 'Benzene'
+        >>> filled = mb.fill_box(benzene,
+        ...                      n_compounds=10,
+        ...                      box=[0, 0, 0, 4, 4, 4])
+        >>> filled.label_rigid_bodies(distinct_bodies='Benzene',
+        ...                           rigid_particles='C')
+
+        """
+        if discrete_bodies is not None:
+            if isinstance(discrete_bodies, string_types):
+                discrete_bodies = [discrete_bodies]
+        if rigid_particles is not None:
+            if isinstance(rigid_particles, string_types):
+                rigid_particles = [rigid_particles]
+
+        if self.root.max_rigid_id is not None:
+            rigid_id = self.root.max_rigid_id + 1
+            warn("{} rigid bodies already exist.  Incrementing 'rigid_id'"
+                 "starting from {}.".format(rigid_id, rigid_id))
+        else:
+            rigid_id = 0
+
+        for successor in self.successors():
+            if discrete_bodies and successor.name not in discrete_bodies:
+                pass
+            else:
+                for particle in successor.particles():
+                    if rigid_particles and particle.name not in rigid_particles:
+                        pass
+                    else:
+                        particle.rigid_id = rigid_id
+                if discrete_bodies:
+                    rigid_id += 1
+
+    def unlabel_rigid_bodies(self):
+        '''Remove all rigid body labels from the Compound '''
+        self._reset_rigid_ids(rigid_id=None)
+
+    def _reset_rigid_ids(self, rigid_id):
+        """Resets the rigid_id of all rigid Particles in a Compound
+
+        Treats the Compound and all successors that already have a rigid_id as
+        a single rigid body.
+        """
+        for particle in self.particles():
+            if particle.rigid_id is not None:
+                if rigid_id is not None:
```

Comment:
I've broken this function into two functions because it was really doing two different things.  In the case of a rigid_id that was not None the function would _add_ that value to the current rigid_id rather than replace it.  I've renamed the function to `_increment_rigid_ids` to reflect this, and renamed the `rigid_id` argument to `increment`, which must be an integer value.  To reset all IDs to None, `unlabel_rigid_bodies` is used.